src/openrct2/util/Util.cpp

```diff
@@ -361,53 +362,59 @@ bool UtilGzipCompress(FILE* source, FILE* dest)
     {
         return false;
     }
-    int ret, flush;
+
     size_t have;
     z_stream strm{};
     strm.zalloc = Z_NULL;
     strm.zfree = Z_NULL;
     strm.opaque = Z_NULL;
-    unsigned char in[CHUNK];
-    unsigned char out[CHUNK];
-    int windowBits = 15;
-    int GZIP_ENCODING = 16;
-    ret = deflateInit2(&strm, Z_DEFAULT_COMPRESSION, Z_DEFLATED, windowBits | GZIP_ENCODING, 8, Z_DEFAULT_STRATEGY);
-    if (ret != Z_OK)
+
+    std::vector<uint8_t> in(InflateDeflateBufferSize);
+    std::vector<uint8_t> out(InflateDeflateBufferSize);
+
+    const int windowBits = 15;
+    const int gzipEncoding = 16;
```

Comment:
```suggestion
    constexpr intwindowBits = 15;
    constexpr int gzipEncoding = 16;
```
Might as well make them `constexpr`?